File: src/main/java/com/stage/teamb/services/address/AddressService.java
package com.stage.teamb.services.address;

import com.stage.teamb.dtos.address.AddressDTO;
import com.stage.teamb.models.Address;

import java.util.List;
import java.util.Optional;

public interface AddressService {

    List<AddressDTO> findAllAddresses();

    AddressDTO findAddressById(Long id);

    AddressDTO saveAddress(AddressDTO addressDTO);

    void deleteAddressById(Long id);

    AddressDTO updateAddress(AddressDTO addressDTO);


    List<AddressDTO> findAddressesByEmployeeId(Long employeeId);

    AddressDTO associateEmployeeWithAddress(Long addressId, Long employeeId);

    AddressDTO disassociateEmployeeFromAddress(Long addressId);

    List<Address> findAll();

    Optional<Address> findOne(Long id);

    Address saveOne(Address address);

    void deleteOne(Long id);

    Boolean existsById(Long id);



}

File: src/main/resources/teamb-front/src/app/shared/services/address.service.ts
import { Injectable, OnDestroy } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable, Subscription } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class AddressService {
 
  private baseUrl = 'http://localhost:8082/api/addresses';

  constructor(private http: HttpClient) { }


  findAllAddresses(): Observable<any[]> {
    const request = this.http.get<any[]>(`${this.baseUrl}/find/all`);
    return request;
  }

  findAddressById(id: number): Observable<any> {
    const request = this.http.get<any>(`${this.baseUrl}/get/${id}`);
    return request;
  }

  saveAddress(addressDTO: any): Observable<any> {
    const request = this.http.post<any>(`${this.baseUrl}/create`, addressDTO);
    return request;
  }

  updateAddress(addressDTO: any): Observable<any> {
    const request = this.http.put<any>(`${this.baseUrl}/update/${addressDTO.id}`, addressDTO);
    return request;
  }

  deleteAddressById(id: number): Observable<void> {
    const request = this.http.delete<void>(`${this.baseUrl}/delete/${id}`);
    return request;
  }

  findAddressesByEmployeeId(employeeId: number): Observable<any[]> {
    const request = this.http.get<any[]>(`${this.baseUrl}/employee/${employeeId}`);
    return request;
  }

  associateEmployeeWithAddress(addressId: number, employeeId: number): Observable<any> {
    const request = this.http.put<any>(`${this.baseUrl}/${addressId}/employee/${employeeId}`, {});
    return request;
  }

  disassociateEmployeeFromAddress(addressId: number): Observable<any> {
    const request = this.http.delete<any>(`${this.baseUrl}/${addressId}/employee`);
    return request;
  }

  findAll(): Observable<any[]> {
    const request = this.http.get<any[]>(`${this.baseUrl}/find/all`);
    return request;
  }

  
}
